### PR TITLE
Move to NAT64 default setup

### PR DESCRIPTION
**Pull request by mike-scott**
- Split out interface-monitor as it's own container so it's more re-usable
- Drop NGINX and move to DNS64/NAT64 stack by default

### CODE REVIEW

**Comment by mike-scott** [comment]
NOTE: This requires a series of patches for the ZmP side.  It's coming soon (tm).

**Comment by mike-scott** [comment]
@doanac My docker-app questions were based on the changes in this PR.

**Comment by mike-scott** [comment]
@doanac I was trying to avoid too much complexity with 1 container.  The reality is we would need to test all the ways a single container could be run.  This way seemed.. cleaner.

**Comment by mike-scott** [comment]
Ok, finally fixed iface-monitor:
- Replaced shell script with janky kill process hacks with a nice clean Python script
- Fixed up usage of /var/run across the board

**Comment by mike-scott** [comment]
@rsalveti please take a look at the very last commit removing tmpfs for /var/run.

File: bt-joiner/start.sh [hunk 1]
```diff
@@ -1,8 +1,7 @@
 #!/bin/sh
 
 # Manually start dbus
-mkdir -p /var/run/dbus
-/usr/bin/dbus-daemon --system
+mkdir -p /var/run/dbus && /usr/bin/dbus-daemon --system
```
Comment (ricardosalveti) [inline bt-joiner/start.sh]: Not an issue, but unrelated change.

File: docker-compose.lwm2m.yml [hunk 2]
```diff
@@ -14,35 +38,44 @@ services:
...
+    network_mode: "host"
+    restart: always
+    read_only: true
+
+  jool:
+    image: ${REGISTRY:-hub.foundries.io}/nat64-jool:${TAG:-latest}
+    tmpfs:
+      - /run
+      - /var/lock
+      - /var/log
```
Comment (ricardosalveti) [inline docker-compose.lwm2m.yml]: Do we need these 3 tmpfs mount points for jool?

File: bt-joiner/start.sh (hunk 1 above)
Comment (mike-scott) [reply bt-joiner/start.sh]: Ah yes, let me remove this change.  Originally I was going to set the mkdir here to /run/dbus.

File: docker-compose.lwm2m.yml (hunk 2 above)
Comment (mike-scott) [reply docker-compose.lwm2m.yml]: Probably don't need any.  Copy/Pasta during setup.

File: docker-compose.lwm2m.yml [hunk 3]
```diff
@@ -14,35 +38,44 @@ services:
...
+      - /var/bind
     ports:
-      - "5683:5683/udp"
-      - "5684:5684/udp"
-      - "5685:5685/udp"
-      - "5686:5686/udp"
-    networks:
-      - gateway-network
+      - "53:53/udp"
```
[the remark this answers is not in the document]
Comment (mike-scott) [reply docker-compose.lwm2m.yml]: Looks like I can use IPv6 notation for this to be:
- "::1:53:53/udp"

[the remark this answers is not in the document]
Comment (mike-scott) [reply docker-compose.lwm2m.yml]: @rsalveti I made this change and then noticed that we shouldn't be setting up "ports" at all (from the docker compose docs):

> **Note:** Port mapping is incompatible with `network_mode: host`

[the remark this answers is not in the document]
Comment (mike-scott) [reply docker-compose.lwm2m.yml]: Now I'm sort of confused.  I don't think every port used in the container is auto-mapped to the host.  How would we specify which ports are?

[the remark this answers is not in the document]
Comment (mike-scott) [reply docker-compose.lwm2m.yml]: I think I can remove network_mode="host" form both DNS64 and californium-proxy containers.  What do you think @rsalveti ?

**Comment by mike-scott** [comment]
Ok, this should be the final update to the PR.  I've tested this configuration with the following notes:
1. For now DNS64 will need to use network_mode: host as there seems to be a bug when specifying ":::53:53/udp" for port mapping.  Docker complains that a binding is already in use (referring to the IPv4 binding of 127.0.0.53:53 that resolved uses.
2. Reverted the removal of network_mode: host from Jool as well.  Turns out the userspace setup gets messed up when that's not present.
3. Re-instated the gateway_network so that coap-proxy can expose it's port correctly and stay away from the host network.
4. Added "/udp" to the EXPOSE statements in coap-proxy and DNS64 so that docker-compose and/or running the container with -P wouldn't create an unnecessary TCP binding.

**Comment by mike-scott** [comment]
Clarifying the use of network_mode: host for DNS64:
- During initialization it looks at what is already bound and makes the needed bindings dynamically (IE: we end up with an [::]:53 binding like we should.)
